scripts/control_pane.js

```diff
@@ -0,0 +1,327 @@
+            fillNavBox([topButton, northButton, "", midButton, hideButton, eastButton, botButton, southButton, ""]);
+            break;
+        case "E":
+            controlLast();
+            controlVert();
+            controlPane.css({borderWidth: "0 0 0 1px"});
+            menu.css({top: "", left: "", right: "3em", bottom: ""});
+            fillNavBox(["", northButton, topButton, westButton, hideButton, midButton, "", southButton, botButton]);
+            break;
+        case "S":
+            controlLast();
+            controlHoriz();
+            controlPane.css({borderWidth: "1px 0 0 0"});
+            menu.css({top: "", left: "", right: "", bottom: "2em"});
+            fillNavBox(["", northButton, "", westButton, hideButton, eastButton, leftButton, centreButton, rightButton]);
+            break;
+        }
+    }
+
+    function navStyle() {
+        switch(begMidEnd) {
+        case "B":
+            control1.css({flex: '0 0 auto'});
+            control3.css({flex: '1 0 auto'});
+            break;
+        case "M":
+            control1.css({flex: '1 0 auto'});
+            control3.css({flex: '1 0 auto'});
+            break;
+        case "E":
+            control1.css({flex: '1 0 auto'});
+            control3.css({flex: '0 0 auto'});
+            break;
+        }
+    }
+
+    hideButton.click(() => {
+        menu.hide();
+    });
+
+    function newPoint(newP) {
+        point = newP;
+        data.compassPoint = point;
+        saveData();
+        navigate();
+        menu.hide();
+    }
+
+    northButton.click(() => {
+        newPoint("N");
+    });
+
+    southButton.click(() =>{
+        newPoint("S");
+    });
+
+    westButton.click(() => {
+        newPoint("W");
+    });
+
+    eastButton.click(() => {
+        newPoint("E");
+    });
+
+    function newBME(newBME) {
+        begMidEnd = newBME;
+        data.begMidEnd = begMidEnd;
+        saveData();
+        navStyle();
+        menu.hide();
+    }
+
+    leftButton.click(() => {
+        newBME('B');
+    });
+
+    centreButton.click(() => {
+        newBME('M');
+    });
+
+    rightButton.click(() => {
+        newBME('E');
+    });
+
+    topButton.click(() => {
+        newBME('B');
+    });
+
+    midButton.click(() => {
+        newBME('M');
+    });
+
+    botButton.click(() => {
+        newBME('E');
+    });
+
+    function addControls(controls) {
+        // put each control in a div
+        controls.forEach(function(control) {
+            control2.append($("<div>").addClass("condiv")
+                .append(control));
+        });
+        // set up the condivs to be block or inline
+        navigate();
+    }
+    // initialise from point NESW and Begin Mid End
+    navigate();
+    navStyle();
+
+    return {
+        container: container,
+        contentPane: contentPane,
+        addControls: addControls
+    };
+}
+
+function makeImagePane(imagePaneData, saveData) {
```

Comment:
Naming-wise, this doesn't have to be a _pane_, it could just be a random `<div>` in the page that the caller wants an image widget in.

Reply:
The same control area could in principle be used with a text area, (though I haven't managed to make it work properly with the sub-splitter in the mentor mode yet).

Reply:
If you want to make this a generic control pane that could be used for both an image and text it needs some restructuring to be more modular. A generic control pane shouldn't include controls other than managing the pane positions. The caller would construct the controls to insert into the control pane (such as the fit-to-size, zoom percent, zoom in/out) and the key to persist the position for that.

I'm not against that work, but if that's your intent it needs some rearchitecting.

Reply:
That should be easy to do. I'm also now inclined to give the divs to the function rather than return them as you suggested earlier.
How about "control bar" for the more generic name?

Reply:
I just noticed your original proposal allowed for persistent fit width or height. I'll make it work like that. We can always change it if people prefer it not to persist.

Reply:
Your way of storing toolbar position is really neat. I'll make it like that.

Reply:
"Control bar" is a great name for the generic dockable control bar.

> That should be easy to do. I'm also now inclined to give the divs to the function rather than return them as you suggested earlier.

Actually, my suggestion for the image widget was to pass the container `<div>`s to the function and have your code consider it the parent bounding box and add the necessary contents inside it. I was unclear about that, my apologies. For the control bar I'm not exactly certain what makes the most sense to pass in such that it can easily be reused.

Reply:
> I just noticed your original proposal allowed for persistent fit width or height. I'll make it work like that. We can always change it if people prefer it not to persist.
This seems to lead to excessive complexity, perhaps leave it to later.

Reply:
That makes sense to me.